src/basic/devnum-util.c

```diff
@@ -57,6 +58,42 @@ int device_path_make_major_minor(mode_t mode, dev_t devnum, char **ret) {
         return 0;
 }
 
+int device_path_make_major_minor2(mode_t mode, dev_t devnum, char **ret) {
+        const char *t;
+        _cleanup_free_ char *syspath = NULL;
+        _cleanup_free_ char *link = NULL;
+        _cleanup_free_ char *fname = NULL;
```

Comment:
please merge these three lines

Reply:
Considered using a single variable. Does `_cleanup_free_` auto free when a new value is assigned?

Reply:
using 3 variables is fine, but declare them on the same line

Reply:
ah :)